src/index.ts

```diff
@@ -6,41 +6,76 @@
  * MIT Licensed
  */
 
-export class Crc32 {
-  #CRC32_POLYNOMIAL = 0xedb88320;
-  #table = new Uint32Array(256).fill(0);
+class InvalidInputError extends Error {
+  constructor(public data: unknown) {
+    super("Invalid input: data must be a Uint8Array.");
```

Comment:
I'd probably add the type of `data` to the message so that the caller knows what the problem is